test/test_helpers.h

```diff
@@ -10,6 +10,7 @@
 #include <math.h>
 #include <float.h>
 
-#define assert_double_equal(a, b) assert_true(fabs(a - b) <= DBL_EPSILON)
+#define assert_double_equal(a, b) assert_true(fabs(a - b) <= DBL_EPSILON * fmax(fabs(a), fabs(b)))
+#define assert_float_equal(a, b) assert_true(fabsf(a - b) <= FLT_EPSILON * fmaxf(fabsf(a), fabsf(b)))
```

Comment:
Macro arguments always need to be "quoted" with ().

Reply:
Done.